drivers/adc/adc_stm32.c

```diff
@@ -716,6 +728,13 @@ static int adc_stm32_init(const struct device *dev)
 	}
 #endif
 
+	// handle internal channels
+#if DT_N_S_soc_S_adc_40012000_P_ch17_internal == 1
```

Comment:
I will leave the SoC specific parts of this to @erwango, but I am -1 on using the generated devicetree_unfixed.h macros directly. They are implementation details. Use devicetree.h macros only please.

Reply:
@mbolivar-nordic Sure. We'll first review the pure driver implementation first (to make sure the proposed changes make sense), then we'll focus on getting the change compliant (and simply functional) vs DT API.

Reply:
fyi
I changed the mentioned lines. Now using dts macros
I also used C comments instead of C++ comments